pandas/core/frame.py

```diff
@@ -4635,7 +4636,16 @@ def quantile(self, q=0.5, axis=0, numeric_only=True):
             0 <= q <= 1, the quantile(s) to compute
         axis : {0, 1, 'index', 'columns'} (default 0)
             0 or 'index' for row-wise, 1 or 'columns' for column-wise
```

Comment:
you put a single doc-string in ``core/generic.py``, with replacements as needed. Then ``Series`` and ``DataFrame`` use that one. The signature should be the same, the ``axis=0`` parameter is asserted to be 0.